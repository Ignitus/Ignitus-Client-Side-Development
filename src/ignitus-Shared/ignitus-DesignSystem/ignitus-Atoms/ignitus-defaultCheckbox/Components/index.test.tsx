import React from 'react';
import { mount, render } from 'enzyme';

import { DefaultCheckbox } from './index';

const props = {
  label: 'Label',
  value: 'Value',
};

it('should rendered', () => {
  const tree = render(<DefaultCheckbox {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultCheckbox />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DefaultCheckbox {...props} />);
  });

  it('should contain checkbox with correct attributes', () => {
    const input = wrapper.find('input[type="checkbox"]');
    expect(input.length).toBe(1);
    expect(input.props().value).toBe(props.value);
    expect(input.props().name).toBe(props.label);
  });

  it('should contain label as a visible UI text', () => {
    expect(wrapper.find('label').props().htmlFor).toBe(props.label);
    expect(wrapper.text()).toBe(props.label);
  });
});
