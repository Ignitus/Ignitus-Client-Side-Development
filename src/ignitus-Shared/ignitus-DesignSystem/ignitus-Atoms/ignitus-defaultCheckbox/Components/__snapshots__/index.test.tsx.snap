// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
<div
  class="css-9ncwx"
>
  <input
    class="css-7sjvxi"
    id="Label"
    name="Label"
    type="checkbox"
    value="Value"
  />
  <label
    class="css-5yljri"
    for="Label"
  >
    <p
      class="css-nbwbpt"
    >
      Label
    </p>
  </label>
</div>
`;
